### PR TITLE
[cisco_ise] Improve handling of unset fields from source

### PR DESCRIPTION
**Pull request by mjwolf**
## Proposed commit message

Improve the handling of unset/null data fields, by checking for null before using the data in processors.

Some log messages do not set all expected fields. This adds more checks that data is not null attempting to run processors on these fields. This also adds some examples of log messages that do not populate some datafields to the pipeline tests.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 1]
```diff
@@ -557,4 +582,5 @@ on_failure:
       value: pipeline_error
   - append:
       field: error.message
-      value: '{{{ _ingest.on_failure_message }}}'
+      value: |-
+        Processor "{{ _ingest.on_failure_processor_type }}" with tag "{{ _ingest.on_failure_processor_tag }}" in pipeline "{{ _ingest.on_failure_pipeline }}" failed with message "{{ _ingest.on_failure_message }}"
```
Comment (andrewkroh) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:586]:
```suggestion
      value: >-
        Processor '{{ _ingest.on_failure_processor_type }}'
        {{#_ingest.on_failure_processor_tag}}with tag '{{ _ingest.on_failure_processor_tag }}'
        {{/_ingest.on_failure_processor_tag}}failed with message '{{ _ingest.on_failure_message }}'
```

For consideration, this omits the "with tag" part if the processor does not have a tag.

Comment (mjwolf) [reply packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml]: thanks, good idea

**Comment by mjwolf** [comment]
I don't Sonarqube is calculating coverage correctly for this, it's saying at least two blocks are uncovered, when testcases exist with data for them (`User-Name` and `cisco-av-pair`)

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 2]
```diff
@@ -124,7 +129,10 @@ processors:
       target_field: cisco_ise.log.calling_station.id
       ignore_missing: true
   - foreach:
-      if: "ctx.cisco_ise?.log?.log_details['cisco-av-pair'] != null && ctx.cisco_ise?.log?.log_details['cisco-av-pair'] instanceof List"
+      tag: foreach_av_pair
+      if: >-
+        ctx.cisco_ise?.log != null && ctx.cisco_ise?.log?.log_details != null &&
+        ctx.cisco_ise?.log?.log_details['cisco-av-pair'] != null && ctx.cisco_ise?.log?.log_details['cisco-av-pair'] instanceof List
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:135]: Can be simplified to:
```yaml
if: ctx.cisco_ise?.log?.log_details != null && ctx.cisco_ise.log.log_details['cisco-av-pair'] instanceof List
```

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 3]
```diff
@@ -138,7 +146,10 @@ processors:
                 value: '{{{_ingest.on_failure_message}}}'
       ignore_missing: true
   - kv:
-      if: "ctx.cisco_ise?.log?.log_details['cisco-av-pair'] != null && !(ctx.cisco_ise?.log?.log_details['cisco-av-pair'] instanceof List)"
+      tag: kv_av_pair
+      if: >-
+        ctx.cisco_ise?.log != null && ctx.cisco_ise?.log?.log_details != null &&
+        ctx.cisco_ise?.log?.log_details['cisco-av-pair'] != null && !(ctx.cisco_ise?.log?.log_details['cisco-av-pair'] instanceof List)
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:152]: Can be simplified to:

```yaml
if: ctx.cisco_ise?.log?.log_details != null && ctx.cisco_ise.log.log_details['cisco-av-pair'] instanceof String
```

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 4]
```diff
@@ -466,27 +481,35 @@ processors:
...
       allow_duplicates: false
       ignore_failure: true
   - append:
+      tag: append_related_user_1
       field: related.user
       value: '{{{cisco_ise.log.log_details.lUser-Name}}}'
-      if: ctx.cisco_ise?.log?.log_details['User-Name'] != null
+      if: >-
+        ctx.cisco_ise?.log != null && ctx.cisco_ise?.log?.log_details != null &&
+        ctx.cisco_ise?.log?.log_details['User-Name'] != null
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:498]: Can be simplified to:

```yaml
if: ctx.cisco_ise?.log?.log_details != null && ctx.cisco_ise.log.log_details['User-Name'] != null
```

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 5]
```diff
@@ -466,27 +481,35 @@ processors:
...
-      if: ctx.cisco_ise?.log?.log_details['User-Name'] != null
+      if: >-
+        ctx.cisci_ise?.log != null && ctx.cisco_ise?.log?.log_details != null &&
+        ctx.cisco_ise?.log?.log_details['User-Name'] != null
       allow_duplicates: false
       ignore_failure: true
   - append:
+      tag: append_related_user_1
       field: related.user
       value: '{{{cisco_ise.log.log_details.lUser-Name}}}'
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:495]: Looks like there's a typo in this line:
```suggestion
      value: '{{{cisco_ise.log.log_details.User-Name}}}'
```

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 6]
```diff
@@ -466,27 +481,35 @@ processors:
       field: cisco_ise.log.log_details.UserName
       ignore_missing: true
   - append:
+      tag: append_user_name_1
       field: user.name
       value: '{{{cisco_ise.log.log_details.User-Name}}}'
-      if: ctx.cisco_ise?.log?.log_details['User-Name'] != null
+      if: >-
+        ctx.cisci_ise?.log != null && ctx.cisco_ise?.log?.log_details != null &&
+        ctx.cisco_ise?.log?.log_details['User-Name'] != null
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:489]: Can be simplified to:

```yaml
if: ctx.cisco_ise?.log?.log_details != null && ctx.cisco_ise.log.log_details['User-Name'] != null
```

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml (hunk 5 above)
Comment (mjwolf) [reply packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml]: thanks, good catch

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml (hunk 2 above)
Comment (mjwolf) [reply packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml]: I've simplified all the conditionals

File: packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml [hunk 7]
```diff
@@ -145,7 +145,7 @@ processors:
       ignore_missing: true
   - kv:
       tag: kv_av_pair
-      if: ctx.cisco_ise?.log?.log_details != null && ctx.cisco_ise.log.log_details['cisco-av-pair'] instanceof String
+      if: ctx.cisco_ise?.log?.log_details != null && !(ctx.cisco_ise.log.log_details['cisco-av-pair'] instanceof List)
```
Comment (taylor-swanson) [inline packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:148]: Why was this changed?

`instanceof String` is what we want here. `!instanceof List` would evaluate true for anything that isn't a List, including null.

Comment (mjwolf) [reply packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml:148]: Sorry, I changed this by changing the first conditions, I missed the last one was different. 

I don't actually know why this processor exists. There isn't a test case that exercises it. Do you know if any real data will be in this format?

Comment (taylor-swanson) [reply packages/cisco_ise/data_stream/log/elasticsearch/ingest_pipeline/pipeline_passed_authentications.yml]: No worries at all.

There might be a historical reason why there's two variations. It wouldn't be the first time that there was two ways of representing data in a Cisco integration, so I'd rather leave it to be safe.

**Review by taylor-swanson — APPROVED** [review]
